tensordict/nn/common.py

```diff
@@ -39,10 +39,25 @@ class FunctionalModuleWithBuffers:
 def _check_all_str(sequence_of_str: Sequence[str]) -> None:
     if isinstance(sequence_of_str, str):
         raise RuntimeError(
-            f"Expected a sequence of strings but got a string: {sequence_of_str}"
+            f"Expected a sequence of strings or tuples of strings, but got: {sequence_of_str}"
         )
-    if any(not isinstance(key, str) for key in sequence_of_str):
-        raise TypeError(f"Expected a sequence of strings but got: {sequence_of_str}")
+    for key in sequence_of_str:
+        if isinstance(key, str):
+            continue
+        elif isinstance(key, tuple):
+            if not len(key):
+                raise RuntimeError("Empty tuples are not permitted.")
+            for sub_key in key:
+                if isinstance(sub_key, str):
+                    continue
+                else:
+                    RuntimeError(
+                        f"Expected a sequence of strings or tuples of strings, but got: {sequence_of_str}"
+                    )
+        else:
+            RuntimeError(
+                f"Expected a sequence of strings or tuples of strings, but got: {sequence_of_str}"
+            )
```

Comment:
We also have the `_check_all_nested` function defined immediately below this which, after these changes, is doing essentially the same thing. 

I took a look, and the only place that `_check_all_str` is used is inside the probabilistic module, so perhaps these two functions can be merged?

Reply:
We need to keep it for the distribution kwargs but we can use the other in all other cases, let me do that